packages/relayer/src/provider-relayer.ts

```diff
@@ -91,7 +104,61 @@ export abstract class ProviderRelayer extends BaseRelayer implements Relayer {
     return (await module.readNonce(space ? space : 0, { blockTag: blockTag })).toNumber()
   }
 
-  wait(_metaTxnId: any, _timeout: number): Promise<any> {
-    throw new Error('Method not implemented.')
+  async wait(metaTxnId: string | SignedTransactions, timeout: number): Promise<providers.TransactionResponse & providers.TransactionReceipt> {
+    if (typeof metaTxnId !== 'string') {
+      console.log("computing id", metaTxnId.config, metaTxnId.context, metaTxnId.chainId, ...metaTxnId.transactions)
+      return this.wait(
+        computeMetaTxnHash(addressOf(metaTxnId.config, metaTxnId.context), metaTxnId.chainId, ...metaTxnId.transactions),
+        timeout
+      )
+    }
+  
+    // Transactions can only get executed on nonce change
+    // get all nonce changes and look for metaTxnIds in between logs
+    const timeoutTime = new Date().getTime() + timeout
+    let lastBlock: number = 0
+
+    const normalMetaTxnId = metaTxnId.replace('0x', '')
+
+    while (new Date().getTime() < timeoutTime) {
+      const block = await this.provider.getBlockNumber()
+      const logs = await this.provider.getLogs({
+        fromBlock: Math.max(0, lastBlock - 12),
+        toBlock: block,
+        // Nonce change event topic
+        topics: ['0x1f180c27086c7a39ea2a7b25239d1ab92348f07ca7bb59d1438fcf527568f881']
```

Comment:
Can we get the topics from the ABI by anychance? Would be a bit cleaner.

Also, can we filter by wallet address? I.e. the contract address that emits this event? Would probably return less logs.

Reply:
We could filter by wallet address, but that's not currently part of the relayer interface, we would need to do a bigger change and add it to the interface

Reply:
Right, we are only passing the metatxnID, I see. Lets keep it for the future, if we see polling without such filter is too strenuous (e.g. if we have 1000s of wallets doing txs at the same time).